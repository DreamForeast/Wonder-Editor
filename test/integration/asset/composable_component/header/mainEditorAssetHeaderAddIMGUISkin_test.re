open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("MainEditorAssetHeader->add imgui skin", () => {
    let sandbox = getSandboxDefaultVal();

    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initState(~sandbox, ());

      MainEditorSceneTool.prepareScene(sandbox);

      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));

    describe("if not select specific treeNode", () =>
      describe("should add into root treeNode", () =>
        test("test snapshot", () => {
          let assetTreeData =
            MainEditorAssetTreeTool.BuildAssetTree.buildEmptyAssetTree();

          MainEditorAssetHeaderOperateNodeTool.addIMGUISkin();

          BuildComponentTool.buildAssetChildrenNode()
          |> ReactTestTool.createSnapshotAndMatch;
        })
      )
    );

    describe("else", () =>
      describe("add into specific treeNode", () =>
        test("test snapshot", () => {
          let assetTreeData =
            MainEditorAssetTreeTool.BuildAssetTree.Folder.TwoLayer.buildOneFolderAssetTree();

          MainEditorAssetTreeTool.Select.selectFolderNode(
            ~nodeId=
              MainEditorAssetTreeTool.BuildAssetTree.Folder.TwoLayer.getFirstFolderNodeId(
                assetTreeData,
              ),
            (),
          );
          MainEditorAssetHeaderOperateNodeTool.addIMGUISkin();

          BuildComponentTool.buildAssetChildrenNode()
          |> ReactTestTool.createSnapshotAndMatch;
        })
      )
    );
    /* describe("test name", () =>
         describe("name should be unique in imgui skin assets", () =>
           test("test add imgui skin asset in different folders", () => {
             let assetTreeData =
               MainEditorAssetTreeTool.BuildAssetTree.Folder.TwoLayer.buildOneFolderAssetTree();

             MainEditorAssetTreeTool.Select.selectFolderNode(
               ~nodeId=
                 MainEditorAssetTreeTool.BuildAssetTree.Folder.TwoLayer.getFirstFolderNodeId(
                   assetTreeData,
                 ),
               (),
             );
             MainEditorAssetHeaderOperateNodeTool.addIMGUISkin();

             let addedNodeId = MainEditorAssetIdTool.getNewAssetId();

             MainEditorAssetTreeTool.Select.selectFolderNode(
               ~nodeId=
                 MainEditorAssetTreeTool.BuildAssetTree.Folder.TwoLayer.getRootNodeId(
                   assetTreeData,
                 ),
               (),
             );
             MainEditorAssetHeaderOperateNodeTool.addIMGUISkin();

             IMGUISkinNodeAssetEditorService.getNodeName(addedNodeId)
             |> StateLogicService.getEditorState
             |> expect
             == (
                  IMGUISkinNodeNameAssetService.getNewName()
                  |> MainEditorAssetNameTool.buildUniqueName
                );
           })
         )
       ); */
  });