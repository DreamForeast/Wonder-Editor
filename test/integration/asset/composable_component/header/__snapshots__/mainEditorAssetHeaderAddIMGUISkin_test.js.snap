// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorAssetHeader->add imgui skin else add into specific treeNode test snapshot 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/imguiSkin.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New IMGUISkin
      </span>
    </div>
  </article>
</article>
`;

exports[`MainEditorAssetHeader->add imgui skin if not select specific treeNode should add into root treeNode test snapshot 1`] = `
<article
  className="wonder-asset-assetChildren"
>
  <article
    className="wonder-asset-fileBox "
    onClick={[Function]}
  >
    <div
      className="box-image"
    >
      <img
        onDragStart={[Function]}
        src="./public/img/imguiSkin.png"
      />
    </div>
    <div
      className="item-text "
    >
      <span>
        New IMGUISkin
      </span>
    </div>
  </article>
</article>
`;
